android/src/main/java/com/pagopa/ioreactnativeintegrity/IoReactNativeIntegrityModule.kt

```diff
@@ -226,31 +259,34 @@ class IoReactNativeIntegrityModule(reactContext: ReactApplicationContext) :
    * - The [challenge] exceeds the size of 128 bytes;
    * - The key attestation generation fails.
    * @param challenge the challenge to be included which has a max size of 128 bytes.
-   * @param keyAlias optional key alias for the generated key pair.
+   * @param keyAlias key alias for the generated key pair.
    * @param promise the React Native promise to be resolved or rejected.
    */
   @ReactMethod
-  fun getAttestation(challenge: String, keyAlias: String?, promise: Promise) {
+  fun getAttestation(challenge: String, keyAlias: String, promise: Promise) {
     Thread {
       try {
         // Remove this block if the minSdkVersion is set to 24
         if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N) {
           ModuleException.UNSUPPORTED_DEVICE.reject(promise)
           return@Thread
         }
-        val alias = keyAlias ?: "attestationKeyAlias"
+        if(keyExists(keyAlias)){
+          ModuleException.KEY_ALREADY_EXISTS.reject(promise)
+          return@Thread
+        }
         val hasStrongBox =
           Build.VERSION.SDK_INT >= Build.VERSION_CODES.P && reactApplicationContext.packageManager.hasSystemFeature(
             PackageManager.FEATURE_STRONGBOX_KEYSTORE
           )
-        val keyPair = generateAttestationKey(alias, challenge.toByteArray(), hasStrongBox)
+        val keyPair = generateAttestationKey(keyAlias, challenge.toByteArray(), hasStrongBox)
```

Comment:
question: is it ok to overwrite a key whith the same alias but which is not hardware backed ? (`getKeyPair` will return `null` in this case)

Reply:
What I meant is that the condition `if (keyExists(keyAlias))` does not pass when a key exists but is not hardware-backed. In such cases, we proceed by passing the existing `keyAlias` to `generateAttestationKey`. However, I might be overlooking something :-)

Reply:
I would argue that this is an edge case because we only manage hardware-backed keys in our app. However I agree that we should not override crypto keys in any case. I applied the suggestion in [0c0362e](https://github.com/pagopa/io-react-native-integrity/pull/20/commits/0c0362e3fd16a55a7b7aeea8b7b2de5769a4025d) and removed the hardware backed check when getting the key.

Reply:
We can handle the case when key exists but it's not hardware backend as: 

```
if (!isKeyHardwareBacked(keyPair.private)) {
          ModuleException.KEY_IS_NOT_HARDWARE_BACKED.reject(promise)
          return@Thread
}
```

Reply:
I think that in this context, the information about wether or not the key is hardware-backed is irrelevant. That check is there only to prevent an unwanted override. Does it makes sense to you?

Reply:
Right, yuo have much more context than I do :-)